### PR TITLE
fix(web): restores compat with older Android devices 🐵

### LINKED ISSUE #9443: bug(web): TypeError: Cannot redefine property: name
Sentry Issue: [KEYMAN-WEB-DX](https://keyman.sentry.io/issues/4379865309/?referrer=github_integration)

```
TypeError: Cannot redefine property: name
  at None (blob:null/4baa49ce-2d04-4277-8a76-d240ad93546e:1:69)
```

It's clearly coming from the predictive-text worker, but... it's not clear at all which _part_ of the worker.  Sentry's information here is pretty sparse.

### PR DESCRIPTION
**Pull request by jahorton**
Fixes #9443.

Turns out that the `es6-shim` polyfill doesn't actually polyfill quite _everything_ we want:
- `Object.values`, `Object.entries` - requires Chrome 54.  Fortunately, it's also very easy to workaround with minimal code restructuring.
- `Array.includes`:  Chrome 47.
- `DOMRect` with constructor:  [Chrome 61](https://developer.mozilla.org/en-US/docs/Web/API/DOMRect#browser_compatibility).
    - This one in particular surprised me, given `getBoundingClientRect` goes as far back as Chrome 18; the gesture engine has components built on this method in particular.

Also disables `keepNames` in the lm-worker's es-build config, as noted here: https://github.com/keymanapp/keyman/pull/9943#discussion_r1384247462

@keymanapp-test-bot skip

### CODE REVIEW

File: android/KMEA/app/src/main/assets/other-polyfills.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+var DOMRect = DOMRect || function (x, y, width, height) {
+  this.x = this.left = x;
+  this.y = this.top = y;
+  this.width = width;
+  this.height = height;
+  this.bottom = y + height;
+  this.right = x + width;
+};
+
+// https://stackoverflow.com/questions/53308396/how-to-polyfill-array-prototype-includes-for-ie8
```
Comment (mcdurdin) [inline android/KMEA/app/src/main/assets/other-polyfills.js:11]:
```suggestion
// https://stackoverflow.com/questions/53308396/how-to-polyfill-array-prototype-includes-for-ie8
// Needed until Chrome 47
```

**Review by mcdurdin — APPROVED** [review]

File: android/KMEA/app/src/main/assets/other-polyfills.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+if(!Array.prototype.includes){
+  //or use Object.defineProperty
+  Array.prototype.includes = function(search){
+   return !!~this.indexOf(search);
+ }
+}
```
Comment (darcywong00) [inline android/KMEA/app/src/main/assets/other-polyfills.js:17]: 1. Is there any syntax required for this polyfill to work with KeymanWeb's ES Modules?

2.
https://github.com/keymanapp/keyman/blob/33eae05cb0fa24e87a6193a8ae0a56060179f873/android/KMEA/build.sh#L95-L96

If this is the only KeymanWeb "asset" not copied from KeymanWeb, I'd probably add a comment about "other-polyfills.js" not needing to be copied.

**Comment by darcywong00** [comment]
Will this fix #9561?

File: android/KMEA/app/src/main/assets/other-polyfills.js (hunk 2 above)
Comment (jahorton) [reply android/KMEA/app/src/main/assets/other-polyfills.js:17]: `other-polyfills.js` is hand-maintained, not being copied from a Node package.  The only "copying" done here was copying the `Array.prototype.includes` polyfill used by the lm-worker into other-polyfills.js.

**Comment by jahorton** [comment]
> Will this fix #9561?

I did nothing to target #9561, but it didn't show up, even when I tried looking for it a bit.  It _has_ been noted to be "on again", "off again" in some sense... I'll try to keep an eye out for the issue arising again and do a similar investigation if and when it does occur.

... I suppose I could try creating an alt-repo, checking out the version-tag in the report, and locally rebuilding in a repo attempt.  Maybe it'll turn something up quickly.

**Comment by jahorton** [comment]
... of _course_ I didn't see it - I've been doing `--debug` Android builds, all of which bypass minification.

Manually editing the --debug build to use KMW's minified `release` products - yep, there's the error.  Gives me something to chase.

File: common/web/lm-worker/build-wrap-and-minify.js [hunk 3]
```diff
@@ -34,7 +34,7 @@ if(MINIFY) {
     sourcemap: 'external',
     sourcesContent: DEBUG,
     minify: true,
-    keepNames: true,
+    keepNames: false, // Do NOT enable - will break under Android 5.0 / Chrome 35 environments!
```
Comment (jahorton) [inline common/web/lm-worker/build-wrap-and-minify.js:37]: An update thanks to https://github.com/evanw/esbuild/issues/3477#issuecomment-1795519157: this is likely a non-issue starting in Chrome 42.

https://caniuse.com/mdn-javascript_builtins_function_name_configurable_true

**Comment by mcdurdin** [comment]
> Turns out that the `es6-shim` polyfill doesn't actually polyfill quite _everything_ we want.

Could you update the description to describe the problem in more detail?

**Pull request merged**